### PR TITLE
Settings connected to workers, and others

### PR DESCRIPTION
**Pull request by marcodeltutto**
- The yaml config file now contains the preset for the automatic and assisted modes. They are written in this format
```
respiratory_rate:
    default: 12
    min: 4
    max: 50
    current: None
```
`default` is the preset default, and the acceptable range is `min`-`max`. `current` gets set inside the configuration dictionary only at runtime, when the user either wants to use the preset itself, or specifies a new one.

- I had trouble in finding the default and min/max values of all parameters in the paper...we will have to check this.

- The settings inputs now have predefined ranges (min/max in the yaml file): the user cannot click + or - if the value will end up outside the ranges.

- Once the uses specifies a setting, this is (i) saved into the config dictionary (in the `current field`) and (ii) sent to the ESP via `esp32.set(param, value)`. In case the serial communication fails here, we should implement a way to go around it, which I haven't done right now.

- The settings gui now shows bigger buttons and fonts (I was not able to match the font in the main UI, sorry)

- From the settings panel, the `Load Preset`, `Close` and `Start Auto/Assist` are now connected.

- Plots are now colored as in the monitors.

- Colors have to be revisited because we have dark bkg for plots and white for text, so it is hard to pick a color, that also works with the alarm red.

### CODE REVIEW

**Comment by carmelopellegrino** [comment]
given the latest PRs #14 and #39 queries for the `all` pseudo-measure should be replaced by `get_all()` calls and the returned dictionary used accordingly

**Comment by marcodeltutto** [comment]
Doing it now

**Comment by marcodeltutto** [comment]
Done, this is updated with master @carmelopellegrino

File: gui/data_handler.py [hunk 1]
```diff
@@ -69,23 +69,46 @@ def esp32_io(self, data_callback):
...
-            vti = float(self._esp32.get("vti"))
-            vte = float(self._esp32.get("vte"))
-
-            # data_callback emits a signal, which is
-            # received by esp32_data_callback, which
-            # then sets the parameters in the DataFiller
-            data_callback.emit('mve', mve)
-            data_callback.emit('vti', vti)
-            data_callback.emit('vte', vte)
+            current_values = {}
```
Comment (carmelopellegrino) [inline gui/data_handler.py]: I think you can just remove line 72, correct?

File: gui/data_handler.py [hunk 2]
```diff
@@ -69,23 +69,46 @@ def esp32_io(self, data_callback):
...
+        from those available in the ESP.
+        '''
+
+        # 1) Calculate Tidal Volume
+        # bpm is respiratory rate [1/minute]
+        # flow is respiratory minute volume [L/minute]
+        # we calculate tidal volume as
+        # volume = flow / bpm
+        if 'bpm' in values and 'flow' in values:
+            values['volume'] = values['flow'] / values['bpm']       # L
```
Comment (carmelopellegrino) [inline gui/data_handler.py]: you can remove 106, right? the variable is going to be overwritten immediately afterwards

File: gui/data_handler.py [hunk 3]
```diff
@@ -109,3 +132,12 @@ def start_io_thread(self):
 
         self._threadpool.start(worker)
 
+    def set_data(self, param, value):
+
+        result = self._esp32.set(param, value)
+
+        if result == self._config['return_success_code']:
```
Comment (carmelopellegrino) [inline gui/data_handler.py]: the if statement can be replaced by a single

```python
return self._config['return_success_code']
```

Comment (carmelopellegrino) [reply gui/data_handler.py]: I edited the previous comment

File: gui/data_handler.py (hunk 1 above)
Comment (marcodeltutto) [reply gui/data_handler.py]: Correct, I'll fix this now

File: gui/data_handler.py (hunk 2 above)
Comment (marcodeltutto) [reply gui/data_handler.py]: Yes sorry, was a leftover a a previous calculation I did. Will do

File: gui/data_handler.py (hunk 3 above)
Comment (marcodeltutto) [reply gui/data_handler.py]: That is correct :)

**Comment by carmelopellegrino** [comment]
I found the following misbehaviours. Here is how to reproduce:
- start the program
- click on `Start Automatic`
- click on `Stop Automatic`
- confirm in the popup window
Then it branches into two:
1. if click on `Start Automatic` again:
- click on `Stop Automatic` is not turned into gray (though not clickable) and only after the delay it becomes red
2. if click on `Start Assisted` (i.e. the other one)
- `Start Automatic` doesn`t turn grey (though again not clickable)
- `Start Assisted` doesn`t turn grey until the delay, then red

Another misbehavior:
- start the program
- click on `Start Automatic`
- click on `Settings`
- go to the `Assisted Mode Settings` tab
- click on `Start Assisted`
- confirmation window asks whether to stop Assisted Mode (running is Automatic)
- if confirm, for a small amount of time one can click on both `Start Assisted` and `Start Automatic` buttons. Depending on the one that is clicked the status goes completely banana, like two Stop buttons being red and clickable

**Comment by carmelopellegrino** [comment]
merging even if there are still issues.
@marcodeltutto will correct later

**Pull request merged**